doc/lsst.daf.butler/formatters.rst

```diff
@@ -139,6 +146,22 @@ You can do this using YAML anchors and references but the preferred approach is
 
 If this approach is used the `StorageClass` Python class created by `StorageClassFactory` will inherit from the specific parent class and not the generic `StorageClass`.
 
+Type converters are specified with a ``converters`` section.
+
+.. code-block:: yaml
+
+    StructuredDataDict:
+      pytype: dict
+      converters:
+        lsst.daf.base.PropertySet: toDict()
+    TaskMetadata:
+      pytype: lsst.pipe.base.TaskMetadata
+      converters:
+        lsst.daf.base.PropertySet: lsst.pipe.base.TaskMetadata.from_metadata
+
+In the first definition, the configuration says that if a ``PropertySet`` object is given then the ``toDict()`` method can be called on it and the returned value will be a `dict`.
+In the second definition a ``PropertySet`` is again specified but this time the ``from_metadata`` class method will be called with the ``PropertySet`` as the first parameter and a ``TaskMetadata`` will be returned.
```

Comment:
This difference between `()` and no `()` confused me a lot when I first saw it and it was not obvious what it means. I had to read everything more than once to figure it out. Now I think that `toDict()` is basically the same as `lsst.daf.base.PropertySet.toDict` (non-bound method). Would it be better to drop `()` and use non-bound method syntax for symmetry?

Reply:
I've been pondering this and the `()` syntax has the advantage that it doesn't lead to duplicating the full class name on both sides:

```yaml
lsst.daf.butler.tests.MetricsExample: exportAsDict()
```
vs
```yaml
lsst.daf.butler.tests.MetricsExample: lsst.daf.butler.tests.MetricsExample.exportAsDict
```
The latter does work and doesn't require any code changes to make it work without this ticket. Maybe that's a good enough argument for not trying to bother with the more succinct form.